src/diffusers/models/adapter.py

```diff
@@ -331,8 +331,8 @@ def __init__(
                 self.body.append(AdapterBlock(channels[i], channels[i], num_res_blocks))
 
         self.body = nn.ModuleList(self.body)
-        # XL has one fewer downsampling
-        self.total_downscale_factor = downscale_factor * 2 ** (len(channels) - 2)
+        # XL has only one downsampling AdapterBlock.
+        self.total_downscale_factor = downscale_factor * 2
```

Comment:
Can you explain why we should do this change? Also why should it be different for SDXL compared to: https://github.com/huggingface/diffusers/blob/d558811b2654ad89c93b3eea10120f3878fec2be/src/diffusers/models/adapter.py#L293 ?

Reply:
The `total_downscale_factor` is the max amount that a T2I-Adapter downscales its input. It is affected by the initial `downscale_factor` and the number of downscaling `AdapterBlock`s. `FullAdapter` has `len(channels) - 1` downscaling blocks, and `FullAdapterXL` has `1` downscaling block, hence the difference between their `total_downscale_factor` calculations. (Note that these are designed to mirror the downscaling patterns in their corresponding UNet architectures.)

The `total_downscale_factor` is used to determine appropriate dimensions for the T2I-Adapter input images (e.g. [here](https://github.com/huggingface/diffusers/blob/d558811b2654ad89c93b3eea10120f3878fec2be/src/diffusers/pipelines/t2i_adapter/pipeline_stable_diffusion_xl_adapter.py#L592-L593)). By over-reporting the `total_downscale_factor` for `FullAdapterXL`, we are forcing unnecessary constraints on the input image dimensions.

Reply:
Makes sense !